triedb/database.go

```diff
@@ -108,7 +108,7 @@ func NewDatabase(diskdb ethdb.Database, config *Config) *Database {
 	if config.HashDB != nil && config.PathDB != nil {
 		log.Crit("Both 'hash' and 'path' mode are configured")
 	}
-	if config.PathDB != nil {
+	if config.PathDB != nil || config.IsVerkle {
```

Comment:
I'm curious why you need this? `config.IsVerkle` should imply `config.PathDB`...

Reply:
did you need it before the rebase and didn't revert that afterwards?

Reply:
> config.IsVerkle should imply config.PathDB...

Yes, exactly. I felt like this function was a good place to put that assumption in. Without this, it was possible to initialize a verkle statedb with hashdb as the backend. I have seen it happen in some tests.

Reply:
> it was possible to initialize a verkle statedb with hashdb as the backend

I checked to see, but I am not able to find anything that would prevent the issue above. There is default triedb config (`VerkleDefaults`) that bundles PathDB and Verkle together, but not all tests use that.